words/double.asm

```diff
@@ -258,19 +258,21 @@ z_m_star_slash: rts
 xt_two_constant:
                 jsr underflow_2
 w_two_constant:
-                jsr w_create
-                jsr w_swap
-                jsr w_comma
-                jsr w_comma
-
-                jsr does_runtime    ; does> turns into these two routines.
+                lda #4
+                sta tmpdsp              ; PFA size 4
+                jsr create_dovar
+                ; double appears on stack as ( lo hi ), i.e. NUXI in memory
+;TODO not sure why we swap to a XINU order for the PFA as opposed to
+; sticking with NUXI by just omitting the two w_swap calls here?
```

Comment:
Do you know if it's important for PFA to be in XINU vs NUXI order here?   The tests still work if you delete the matching `swap` but maybe something else relies on it?

Reply:
The forth standard gives specific ordering for how doubles are stored in memory:
[https://forth-standard.org/standard/core/TwoFetch](https://forth-standard.org/standard/core/TwoFetch)

Reply:
ok, it looks like `2!` and `2@` match the standard, but it seems like both 2constant and 2literal use non-standard ordering for the data in their PFA.  

I guess it doesn't matter since it's internal to those routines, but it seems like it'd be easier to understand if they used the standard memory layout like 2!/2@ ?  
 
```
hex  ok
2variable 2foo  ok
12345678. .s <2> 5678 1234  ok  \ high word is TOS, low word is NOS
$74 4 dump 
0074  34 12 78 56  4.xV ok  \ so memory layout of the stack is actually NUXI
2foo 2!  2foo 4 dump 
080F  34 12 78 56  4.xV ok \ 2! also uses NUXI memory layout, matching the standard

12345678. 2constant bar  ok \ but ... 
see bar 
...
081E  20 E1 9F 78 56 34 12   ..xV4.  \ 2constant stores its PFA in XINU order
...
0 nc-limit !  ok
: foobar 12345678. ;  ok
see foobar 
...
0833  20 7C A2 78 56 34 12   |.xV4.  \ ... as does 2literal

833   A27C jsr     2LITERAL 12345678 
 ok
```

Reply:
It looks like 2constant and 2literal are using the string handling code, which compiles "addr len" into memory in that order.

Absolutely could swap it around for consistency with the double words, but strings and disassembler (which knows about strings) will need to be updated.

Does it make any sense to save it in stack order, so a 4-cycle loop can be used to shuffle directly from RAM onto stack in the same order?  That might provide some speed/space improvements.

Reply:
Ah that makes sense.  Doing them all the same way with stack order matching the standard and the pfa seems nice and might be simpler/faster like you say.

Reply:
maintaining stack order as you suggest does seem better, see #123